p2p/addressbook/store.go

```diff
@@ -169,19 +197,18 @@ func (a *AddrBook) decodeFrom(path string) error {
 func (a *AddrBook) Persist(ctx context.Context) {
 	ticker := time.NewTicker(persistInterval)
 	defer ticker.Stop()
-	path := a.path
-	if len(path) == 0 {
+	if len(a.path) == 0 {
 		return
 	}
 
 	for {
 		select {
 		case <-ticker.C:
-			a.persistPeers(path)
-			a.logger.Debug("saved peers to file %v", path)
+			a.persistPeers()
+			a.logger.Debug("saved peers to file %v", a.path)
```

Comment:
i think you need to at least return the error from persistPeers to know if it's successful before logging this